cms/djangoapps/contentstore/views/item.py

```diff
@@ -382,15 +383,16 @@ def xblock_view_handler(request, usage_key_string, view_name):
 
             # Set up the context to be passed to each XBlock's render method.
             context = {
-                'is_pages_view': is_pages_view,     # This setting disables the recursive wrapping of xblocks
+                # This setting disables the recursive wrapping of xblocks
+                'is_pages_view': is_pages_view or view_name == AUTHOR_VIEW,
                 'is_unit_page': is_unit(xblock),
                 'can_edit': can_edit,
                 'root_xblock': xblock if (view_name == 'container_preview') else None,
                 'reorderable_items': reorderable_items,
                 'paging': paging,
                 'force_render': force_render,
             }
-
+            context.update(request.GET.dict())
```

Comment:
Is it safe to just splat the querystring into the context like this?

Reply:
Hmm, I suppose that somebody who's logged in could override `can_edit` and receive a fragment for editing (but they won't be able to save it).

I could move the `context.update` up to avoid overriding any of the critical keys in the dict.

Reply:
That works. Or grabbing explicit keys. I'm mostly worried that even if this is safe to do now, it might become unsafe as the code is innocently modified a year or two down the line and new things are added to the context.